### DISCUSSION TITLE
Welcome to PeriLab.jl Discussions!

### DISCUSSION THREAD

**Discussion by PeriHub** [PeriHub/PeriLab.jl#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by nomiag9** [comment]
Hello Mr. Willberg and Mr. Hesse,
I wanted to ask and also confirm whether Shell elements (S4/R or S8/R) are supported in PeriLab, as I have modeled CFRP specimen using shell element?
Regards

![Annotation 2024-12-04 172845](https://github.com/user-attachments/assets/43e6d1cf-2d50-466e-a734-db0ddd3bc513)

**Reply by JTHesse** [reply]
Hi @nomiag9, as stated in the Documentation we are using the [Abaqusreader.jl package](https://github.com/JuliaFEM/AbaqusReader.jl). Take a look at the elements they are supporting. You could try to change the element name in the abaqus .inp file to C3D4, but I’m uncertain if the third dimension will give you some problems. I’m currently out of Office and aren’t able to check this.

If you want a long time solution you will have to make a feature request for the Abaqusreader.jl package, and let us know.

Alternatively, you could also try to generate the peridynamic mesh by your own.

**Comment by CWillberg** [comment]
Dear Noamaan,

 

PeriLab does allow plane stress, plane strain and 3D formulations. No PD shell approach is implemented yet. If you need such feature, please address it as a feature request. It will take some time, but from the software architecture point of view it should be possible.

 

Christian

 

Von: nomiag9 ***@***.***> 
Gesendet: Mittwoch, 4. Dezember 2024 17:32
An: PeriHub/PeriLab.jl ***@***.***>
Cc: Subscribed ***@***.***>
Betreff: Re: [PeriHub/PeriLab.jl] Welcome to PeriLab.jl Discussions! (Discussion #1)

 

Hello Mr. Willberg and Mr. Hesse,
I wanted to ask and also confirm whether Shell elements (S4/R or S8/R) are supported in PeriLab, as I have modeled CFRP specimen using shell element?
Regards
Noamaan
Annotation.2024-12-04.172845.png (view on web) <https://github.com/user-attachments/assets/43e6d1cf-2d50-466e-a734-db0ddd3bc513>

**Reply by nomiag9** [reply]
Thank you Mr. Willberg for your answer. 
I see that reduced integration is also not supported for the elements. For modeling the interface layer in Abaqus, I would be using cohesive elements with traction separation law but as cohesive elements are also not supported in PeriLab, could you suggest me which other possibility I have for modeling debonding between two different materials with different element type?
Thank you for your cooperation.
Best Wishes

**Comment by CWillberg** [comment]
You can use interface damage models. In that case bonds which run from one block the another can be defined with a different damage value. Its called interblock damage

 

An example can be found here.

https://github.com/PeriHub/PeriLab.jl/blob/main/examples/Undulations/Smetana.yaml

 

This example is a multilayer laminate.

 

Christian

 

Von: nomiag9 ***@***.***> 
Gesendet: Donnerstag, 5. Dezember 2024 10:55
An: PeriHub/PeriLab.jl ***@***.***>
Cc: Christian Willberg ***@***.***>; Comment ***@***.***>
Betreff: Re: [PeriHub/PeriLab.jl] Welcome to PeriLab.jl Discussions! (Discussion #1)

 

Thank you Mr. Willberg for your answer.
I see that reduced integration is also not supported for the elements. For modeling the interface layer in Abaqus, I would be using cohesive elements with traction separation law but as cohesive elements are also not supported in PeriLab, could you suggest me which other possibility I have for modeling debonding between two different materials with different element type?
Thank you for your cooperation.
Best Wishes

**Comment by xibaiw** [comment]
Hello Mr. Willberg and Mr. Hesse,
I did not find any description about pre-cracks in perilab documentation. I want to know if Perilab supports pre-cracks. In which section can I find it？

**Comment by CWillberg** [comment]
Hello,

you can define pre crack via the bond filter. You can define planes, which does not allow bonds running through them.
Here is the documentation
https://perihub.github.io/PeriLab.jl/stable/man/bond_filter/
An an example can be found here
https://github.com/PeriHub/PeriLab.jl/blob/main/test/fullscale_tests/test_DCB/DCBmodel_PD_solid.yaml

**Reply by xibaiw** [reply]
Thank you Mr. Willberg for your prompt response. I will try it right away.
